sqlmesh/core/context.py

```diff
@@ -1318,7 +1319,14 @@ def table_diff(
             if not on:
                 for ref in model.all_references:
                     if ref.unique:
-                        on = ref.columns
+                        expr = ref.expression
+
+                        if isinstance(expr, exp.Tuple):
+                            on = [key.this.sql() for key in expr.expressions]
+                        elif isinstance(expr, exp.Paren):
+                            on = [expr.this.this.sql()]
```

Comment:
Should this be just `expr.this.sql()`?

Reply:
Yes didn't realise it you're right I'm changing this one now thanks